GNUmakefile

```diff
@@ -139,6 +139,7 @@ deps:  ## Install build and development dependencies
 	go install golang.org/x/tools/cmd/stringer@v0.30.0
 	go install github.com/hashicorp/hc-install/cmd/hc-install@v0.9.0
 	go install github.com/shoenig/go-modtool@v0.2.0
+	go install github.com/vektra/mockery/v3@v3.5.1
```

Comment:
Not having a mockable interface for `shoenig/test` means that leaning on this tool will prevent us from finally getting rid of `testify` (which has some fairly serious correctness bugs we've encountered). Which isn't a deal-breaking but it'd be nice if we could start thinking of a path that would let us use a useful tool but still ditch `testify`.

That being said, we've mostly leaned on stub implementations of interfaces rather than mocking of complex mocking frameworks in Nomad previously because the bang-per-buck isn't that high. It's not super obvious that's not the case for the mock consumers here... see my comments on the other two PRs. But I'm happy to :shrug: and see how it goes.

Reply:
I use mockery because of past experience and it was the least friction to throw in and get working. I took a swing at spinning out some lightweight mocks instead (leaning on `shoenig/test`) so I've backed the mockery stuff out of here.